cpp/examples/interop/interop.cu

```diff
@@ -0,0 +1,166 @@
+  out.ref = {size, {}, buffer_index, offset};
+  memcpy(&out.ref.prefix, data, sizeof(out.ref.prefix));
+  return out;
+}
+inline arrow::BinaryViewType::c_type ToStringView(std::string_view v,
+                                                  int32_t buffer_index,
+                                                  int32_t offset)
+{
+  return ToStringView(v.data(), static_cast<int32_t>(v.size()), buffer_index, offset);
+}
+
+/**
+ * @brief Create a StringViewArray
+ *
+ * @param data_buffers The data buffers
+ * @param views The string views
+ * @param validate Whether to validate the array
+ */
+arrow::Result<std::shared_ptr<arrow::StringViewArray>> MakeStringViewArray(
+  arrow::BufferVector data_buffers,
+  const std::vector<arrow::BinaryViewType::c_type>& views,
+  bool validate = true)
+{
+  auto length = static_cast<int64_t>(views.size());
+  auto arr    = std::make_shared<arrow::StringViewArray>(
+    arrow::utf8_view(), length, arrow::Buffer::FromVector(views), std::move(data_buffers));
+  if (validate) { RETURN_NOT_OK(arr->ValidateFull()); }
+  return arr;
+}
+
+auto make_chars_and_sizes(std::vector<std::string> strings)
+{
+  std::vector<char> chars{};
+  std::vector<cudf::size_type> sizes;
+  for (auto& str : strings) {
+    chars.insert(chars.end(), std::cbegin(str), std::cend(str));
+    sizes.push_back(str.length());
+  }
+  return std::make_tuple(std::move(chars), std::move(sizes));
+};
+
+/**
+ * @brief Convert an Arrow StringViewArray to a cudf column
+ *
+ * @param array The Arrow StringViewArray
+ * @param stream The CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ */
+std::unique_ptr<cudf::column> ArrowStringViewToCudfColumn(
+  std::shared_ptr<arrow::StringViewArray> array,
+  rmm::cuda_stream_view stream      = cudf::get_default_stream(),
+  rmm::device_async_resource_ref mr = rmm::mr::get_current_device_resource())
+{
+  // Convert the string views into chars and sizes
+  std::vector<std::string> strings;
+  for (auto i = 0; i < array->length(); i++) {
+    strings.push_back(array->GetString(i));
+  }
+  auto [chars, sizes] = make_chars_and_sizes(strings);
+
+  // Copy the chars vector the device
+  rmm::device_uvector<char> d_chars(chars.size(), stream, mr);
+  CUDF_CUDA_TRY(cudaMemcpyAsync(
+    d_chars.data(), chars.data(), chars.size() * sizeof(char), cudaMemcpyDefault, stream.value()));
+
+  // Copy the sizes vector to the device
+  thrust::device_vector<cudf::size_type> d_sizes(sizes);
+  rmm::device_uvector<cudf::size_type> string_sizes(array->length() + 1, stream);
+  string_sizes.set_element(0, 0, stream);
+  thrust::copy(d_sizes.begin(), d_sizes.end(), string_sizes.begin() + 1);
+
+  // Perform an inclusive scan on the sizes vector to
+  // generate the offsets and wrap it as a cudf column
+  rmm::device_uvector<cudf::size_type> d_offsets(array->length() + 1, stream);
+  thrust::inclusive_scan(
```

Comment:
I think you can use `exclusive-scan` instead of `inclusive-scan` to avoid the extra call to set the first element to 0 and the extra copy of the sizes. Also, see my other comment about doing this in host code instead and just copying the offsets to device memory at the end.